core/local/SyncDir.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+
+var fs = require('fs')
+
+/*::
+import type EventEmitter from 'events'
+*/
+
+module.exports = {
+  ensureExistsSync,
+  startIntervalCheck
+}
+
+/** Make sure syncPath actually exists.
+ *
+ * In case it doesn't, emit 'syncdir-unlinked' and throws.
+ * Any other error occuring during the check will be thrown too.
+ */
+function ensureExistsSync ({syncPath, events} /*: {syncPath: string, events: EventEmitter} */) /*: void */ {
+  if (!fs.existsSync(syncPath)) {
+    events.emit('syncdir-unlinked')
+    throw new Error('Syncdir has been unlinked')
```

Comment:
I kept the lowercase `d` in the error message because changing it would break GUI... :sob: 
I would be nice to refactor error handling globally (error codes maybe) in order not to depend on hardcoded error messages everywhere...